src/server/bg_services/replication_scanner.js

```diff
@@ -0,0 +1,283 @@
+
+            keep_listing_dst = diff.keep_listing_dst;
+            src_contents_left = diff.src_contents_left;
+            ans.keys_to_copy = ans.keys_to_copy.concat(diff.to_replicate_arr);
+
+            // advance dst token only when cur dst list could not contains next src list items
+            const last_dst_key = dst_list.Contents[dst_list.Contents.length - 1].Key;
+            if (last_src_key >= last_dst_key) {
+                new_dst_cont_token = dst_list.NextContinuationToken;
+            }
+        }
+        return {
+            ...ans,
+            // if src_list cont token is empty - dst_list cont token should be empty too
+            dst_cont_token: (src_list.NextContinuationToken && new_dst_cont_token) || ''
+        };
+    }
+
+    async move_objects(copy_type, src_bucket_name, dst_bucket_name, keys) {
+        try {
+            const res = await this._scanner_sem.surround_count(keys.length,
+                async () => {
+                    try {
+                        const res1 = await this.client.replication.move_objects_by_type({
+                            copy_type,
+                            src_bucket_name,
+                            dst_bucket_name,
+                            keys
+                        }, {
+                            auth_token: auth_server.make_auth_token({
+                                system_id: system_store.data.systems[0]._id,
+                                account_id: system_store.data.systems[0].owner._id,
+                                role: 'admin'
+                            })
+                        });
+                        return res1;
+                    } catch (err) {
+                        // no need to do retries, eventually the object will be uploaded
+                        // TODO: serious error codes with metrics (auth_failed, storage not exist etc)
+                        dbg.log0('replication_server move_objects: error: ', err, src_bucket_name, dst_bucket_name, keys);
+                    }
+                });
+            return res;
+        } catch (err) {
+            dbg.error('replication_server move_objects: semaphore error:', err, err.stack);
+            // no need to handle semaphore errors, eventually the object will be uploaded
+        }
+    }
+
+    async list_objects(bucket_name, prefix, continuation_token) {
+        try {
+            dbg.log0('replication_server list_objects: params:', bucket_name, prefix, continuation_token);
+            const list = await this.noobaa_connection.listObjectsV2({
+                Bucket: bucket_name.unwrap(),
+                Prefix: prefix,
+                ContinuationToken: continuation_token,
+                MaxKeys: Number(process.env.REPLICATION_MAX_KEYS) || 1000
+            }).promise();
+
+            dbg.log0('replication_server.list_objects: finished successfully', list.Contents.map(c => c.Key));
```

Comment:
I think that most of the log messages here should be log1. this message for example can print a long message. 

in general go over the `dbg.log0` and see what can be changed. also change error messages to `dbg.error`

Reply:
fixed